BPA/Ch.03/GLM_Binomial.stan

```diff
@@ -0,0 +1,39 @@
+data {
+  int<lower=0> nyears;          // Number of Years
+  int<lower=0> C[nyears];       // Counts
+  int<lower=0> N[nyears];       // Binomial Totals
+  vector[nyears] year;          // Year covariates
+}
+
+parameters {
+  real alpha;
+  real beta1;
+  real beta2;
+}
+
+transformed parameters {
+  vector[nyears] logit_p;
+
+  // Linear predictor
+  logit_p <- alpha +
+             beta1 * year +
+             beta2 * year .* year;
+}
+
```

Comment:
I think I lost my last comment like this, but here's another case.

Given that `year` is data, you want to compute `year .* year` as transformed data and assigned it to a new vector `year_squared`.  And then use `beta2 * year_squared* instead of `beta2 * year .* year` --- it saves a lot of compution to move operations to transformed data when possible.  Here, Stan is left associative, so that the long form parses as `(beta2 * year) .* year`, which introduces a lot of unnecessary autodiff expressions compared to the single multiplication by `year_squared`.